### PR TITLE
Add oneClickUpsell A/B test

### PR DESCRIPTION
**Pull request by taggon**
#### Changes proposed in this Pull Request

* Adds oneClickUpsell A/B test. See pcbrnV-aX-p2 for more details. 

#### Testing instructions

* Make sure you're assigned to `test` group of `oneClickUpsell` abtest.
* Purchase Personal or Premium dotcom plan for your website.
* You may need to remember the receipt ID.
* Visit `/checkout/offer-quickstart-session/RECEIPT_ID/YOUR_SITE_SLUG`. Please replace RECEIPT_ID and YOUR_SITE_SLUG with your owns. It would look like `/checkout/offer-quickstart-session/1234567/testblog.woordpress.com`.
* Ensure you have one or more stored credit cards.
* Click the "Yes, I want a WordPress Expert by my side!" button.
* You should be able to see a modal dialog as below:
  Checkout_‹_Quick_Start_Session_—_WordPress_com
* Click on the Pay button and the transaction of purchasing Concierge Session is going to be _automatically_ processed.
* When everything goes okay, a sucess notification appears on the homepage of your site. Otherwise, an error notification appears.
  My_Home_‹_Site_Title_—_WordPress_com

### CODE REVIEW

**Comment by taggon** [comment]
Please don't mind the conflicts for now. I will fix it once the code fulfilled all the requirements.

**Comment by taggon** [comment]
Rebased.

**Comment by southp** [comment]
I have tested both the control and test variations, and they both works as expected for me. 👍 
@taggon Could you confirm for me if that the plan upsell isn't included is expected? e.g. the upsell for the business plan when I am purchasing a Premium plan.

**Comment by taggon** [comment]
> Could you confirm for me if that the plan upsell isn't included is expected? e.g. the upsell for the business plan when I am purchasing a Premium plan.

@southp The hypothesis of the a/b test can answer that. pbxNRc-sh-p2

> If we offer the user an easier way to purchase a QS session by making an instant transaction, then ARPU will increase and refund rates will maintain or lower.

File: client/my-sites/checkout/upsell-nudge/index.jsx [hunk 1]
```diff
@@ -180,10 +192,65 @@ export class UpsellNudge extends React.Component {
...
+		if ( 'accept' !== buttonAction || 'concierge-quickstart-session' !== upsellType ) {
+			return false;
+		}
+
+		// stored cards should exist
+		if ( cards.length === 0 ) {
+			return false;
+		}
+
+		return 'test' === abtest( 'oneClickUpsell' );
```
Comment (niranjan-uma-shankar) [inline client/my-sites/checkout/upsell-nudge/index.jsx:221]: If the user has two sites, and one of the sites has a stored card, then there's a scenario in which the purchase model could get stuck. To reproduce:

1. On an account that has two sites, purchase a paid plan on one of the sites using a credit card so that your card gets saved.
2. Visit `/checkout/offer-quickstart-session`. Users would reach this link either via the Quick Start upsell banner in `/me/purchases`, or via HEs upselling a Quick Start session via chat or tickets.
3. Click on the "Yes" button on the upsell and notice the purchase model stuck while loading(assuming you get assigned to the variant, else manually assign yourself to the variant and try again).


signup

**Comment by dzver** [comment]
I tested the test and control and also tested to see what happens if there's no stored credit card. It worked as expected.

I saw that @niranjan-uma-shankar found an edge case, but apart from that, I didn't see any issues and the change LGTM. It might be good to also have a Payments person have a look.

Good job!

**Review by niranjan-uma-shankar — COMMENTED** [review]
1. I see that a few new "translate" calls have been added. It would be nice if we can inform @Automattic/i18n to deprioritize these strings in the Weekly kit, so that we don't need to incur translation costs for an `EN` only test until we know the test outcome.

2. On the purchase model, the ToS doesn't appear to be correct. 
" _authorize your payment method to be charged on a recurring basis until you cancel_" seems to indicate that the Quick Start purchase will be a recurrent charge.

Screenshot 2020-09-23 at 3 52 01 PM

For reference, the Quick Start purchase ToS on the full page checkout is below, and I think we should use that here:

Screenshot 2020-09-23 at 3 52 36 PM

**Comment by taggon** [comment]
> 2. On the purchase model, the ToS doesn't appear to be correct.
" authorize your payment method to be charged on a recurring basis until you cancel" seems to indicate that the Quick Start purchase will be a recurrent charge.

Can you check this @victorespigares ?

File: client/my-sites/checkout/upsell-nudge/index.jsx (hunk 1 above)
Comment (taggon) [reply client/my-sites/checkout/upsell-nudge/index.jsx:225]: Thanks @niranjan-uma-shankar ! I think the bug occurs due to lack of the site slug which the modal component depends on. Now the test is disabled when no site is selected. Can you check it again?

**Review by niranjan-uma-shankar — COMMENTED** [review]
A couple more things:

1. When I purchase a Quick Start stand-alone i.e not via the post signup upsell but via the /me/purchases banner, then I am taken to a Thank You page if I'm in `control` which comes with a link to schedule my session and instructions to check email:

Screenshot 2020-09-23 at 4 41 17 PM

But if I'm in the variant, I am taken to "My Home". Any reason why the post purchase destinations are different?

Screenshot 2020-09-23 at 4 41 17 PM

2. If I purchase via the post-signup upsell, then I'm taken to "My Home" with two success messages. This is probably not a big deal, but wanted to note it nevertheless:

Screenshot 2020-09-23 at 4 47 10 PM

**Comment by victorespigares** [comment]
> > 1. On the purchase model, the ToS doesn't appear to be correct.
> >    " authorize your payment method to be charged on a recurring basis until you cancel" seems to indicate that the Quick Start purchase will be a recurrent charge.
> 
> Can you check this @victorespigares ?

Yeah, nice catch. This ToS is likely coming from a Plan subscription which doesn't apply for a QS purchase. Although we're going to be testing it for just QS to begin, the scope is to make the 1-click modal agnostic of the product so we can reuse it. Can we copy the regular Checkout logic for displaying the right ToS for each product here? I'm thinking in QS, but also Plan bumps (like the Premium to Business).

cc: @taggon

**Review by dzver — APPROVED** [review]
LGTM

I think these concierge sessions could benefit from a thank-you screen. Seeing the green success notice is confusing to me as a user, I don't quite know what's supposed to happen.

Also tested buying two, worked as expected.

**Comment by taggon** [comment]
Now the checkout terms section uses the same logic with the checkout page. It will change depends on the selected product.
cc @victorespigares

**Comment by taggon** [comment]
@victorespigares Should we show Thank you page after the one click purchase completes? I'd like to hear your opinion on what @niranjan-uma-shankar and @dzver raised above.

**Comment by victorespigares** [comment]
> A couple more things:
> 
> 1. When I purchase a Quick Start stand-alone i.e not via the post signup upsell but via the /me/purchases banner, then I am taken to a Thank You page if I'm in `control` and this page comes with a link to schedule my session and instructions to check email:
> 
> Screenshot 2020-09-23 at 4 41 17 PM
> 
> But if I'm in the variant, I am taken to "My Home". Any reason why the post purchase destinations are different for control and variant?
> 
> Screenshot 2020-09-23 at 4 44 31 PM
> 
> 1. If I purchase via the post-signup upsell, then I'm taken to "My Home" with two success messages. This is probably not a big deal, but wanted to note it nevertheless:
> 
> Screenshot 2020-09-23 at 4 47 10 PM

- To the point #1, yep, the one-click shouldn't override the original destination. That means that if you're in the Signup flow, will end up in the Customer Home, while if you're in a different part, it might be different. But the one-click should follow the same thank-you destination than the regular checkout.

- To the point #2, I think is fine like that. Both messages refer to different things and we should acknowledge the purchase went through. That doesn't mean we can do better of course. But for now, I think is fine.

- To @dzver's proper thank-you point, I agree, although I'd say that's out of the scope of this project.

**Comment by taggon** [comment]
The latest commit should resolve @niranjan-uma-shankar 's first concern. The one click upsell never overrides the original destination. :)
cc @victorespigares

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4608364

Thank you @taggon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4608364).
